lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3114,18 +3106,21 @@ public function getSingleIdentifierValue($entity)
      * Tries to find an entity with the given identifier in the identity map of
      * this UnitOfWork.
      *
-     * @param mixed  $id            The entity identifier to look for.
-     * @param string $rootClassName The name of the root class of the mapped entity hierarchy.
-     * @psalm-param class-string $rootClassName
+     * @param mixed                 $id        The entity identifier to look for.
+     * @param Mapping\ClassMetadata $rootClass The name of the root class of the mapped entity hierarchy.
      *
      * @return object|false Returns the entity with the specified identifier if it exists in
      *                      this UnitOfWork, FALSE otherwise.
      */
-    public function tryGetById($id, $rootClassName)
+    public function tryGetById($id, $rootClass)
```

Comment:
Changing the signature of a public method is a BC break. We cannot do that without a proper deprecation strategy.

Reply:
let me make a separate PR for that